hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1923,12 +1923,23 @@ private void processAssignQueue() {
       LOG.debug("Processing assignQueue; systemServersCount=" + serversForSysTables.size() +
           ", allServersCount=" + servers.size());
       processAssignmentPlans(regions, null, systemHRIs,
-          serversForSysTables.isEmpty()? servers: serversForSysTables);
+          serversForSysTables.isEmpty() && !containsBogusAssignments(regions, systemHRIs) ?
+              servers: serversForSysTables);
     }
 
     processAssignmentPlans(regions, retainMap, userHRIs, servers);
   }
 
+  private boolean containsBogusAssignments(Map<RegionInfo, RegionStateNode> regions, List<RegionInfo> hirs){
+    for (RegionInfo ri : hirs) {
+      if (regions.get(ri).getRegionLocation() != null &&
+          regions.get(ri).getRegionLocation().equals(LoadBalancer.BOGUS_SERVER_NAME)){
```

Comment:
File an issue to purge this whole BOGUS_SERVER_NAME thing. It is just wrong, especially now rsgroup is integrated into core.

Thanks.

Reply:
Maybe it'll be replaced by choosing a random server in DEFAULT group. There is something wrong when no server can be chosen for group regions in current codes.

Reply:
Ok. Separate issue though I'd say. Others might be interested in the discussion.

Thanks.